bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/RenaultHandler.java

```diff
@@ -0,0 +1,147 @@
+
+        // Validate configuration
+        if (this.config.myRenaultUsername.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "MyRenault Username is empty!");
+            return;
+        }
+        if (this.config.myRenaultPassword.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "MyRenault Password is empty!");
+            return;
+        }
+        if (this.config.locale.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Location is empty!");
+            return;
+        }
+        if (this.config.vin.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "VIN is empty!");
+            return;
+        }
+        if (this.config.refreshInterval < 1) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "The refresh interval mush to be larger than 1");
+            return;
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Background initialization:
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(this::getStatus, 0, config.refreshInterval, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        super.dispose();
+    }
+
+    private void getStatus() {
+        MyRenaultHttpSession httpSession = new MyRenaultHttpSession(this.config, httpClient);
+        try {
+            httpSession.initSesssion();
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            httpSession = null;
+            logger.warn("Error My Renault Http Session.", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+        if (httpSession != null) {
+            httpSession.updateCarData();
+            updateState(httpSession.getCar());
+        }
+    }
+
+    private void updateState(Car car) {
+        if (car.batteryLevel != null) {
+            updateState(CHANNEL_BATTERY_LEVEL, new DecimalType(car.batteryLevel));
+        }
+        if (car.hvacstatus != null) {
+            updateState(CHANNEL_HVAC_STATUS, OnOffType.from(car.hvacstatus));
+        }
+        if (car.imageURL != null && !car.imageURL.isEmpty()) {
+            updateState(CHANNEL_IMAGE, new StringType(car.imageURL));
+        }
+        if (car.gpsLatitude != null && car.gpsLongitude != null) {
+            updateState(CHANNEL_LOCATION,
+                    new PointType(new DecimalType(car.gpsLatitude), new DecimalType(car.gpsLongitude)));
+        }
+        if (car.odometer != null) {
+            updateState(CHANNEL_ODOMETER, new QuantityType<Length>(Double.valueOf(car.odometer), KILO(METRE)));
```

Comment:
Here is the full method to avoid compilation warnings (NPE):
```
    private void updateState(Car car) {
        Double batteryLevel = car.batteryLevel;
        if (batteryLevel != null) {
            updateState(CHANNEL_BATTERY_LEVEL, new DecimalType(batteryLevel.doubleValue()));
        }
        Boolean hvacstatus = car.hvacstatus;
        if (hvacstatus != null) {
            updateState(CHANNEL_HVAC_STATUS, OnOffType.from(hvacstatus.booleanValue()));
        }
        String imageURL = car.imageURL;
        if (imageURL != null && !imageURL.isEmpty()) {
            updateState(CHANNEL_IMAGE, new StringType(imageURL));
        }
        Double latitude = car.gpsLatitude;
        Double longitude = car.gpsLongitude;
        if (latitude != null && longitude != null) {
            updateState(CHANNEL_LOCATION,
                    new PointType(new DecimalType(latitude.doubleValue()), new DecimalType(longitude.doubleValue())));
        }
        Double odometer = car.odometer;
        if (odometer != null) {
            updateState(CHANNEL_ODOMETER, new QuantityType<Length>(odometer.doubleValue(), KILO(METRE)));
        }
    }
```